src/utils/makeReactNativeConfig.js

```diff
@@ -270,4 +289,20 @@ function injectPolyfillIntoEntry(
   return userEntry;
 }
 
-module.exports = { makeReactNativeConfig, injectPolyfillIntoEntry };
+function createWebpackConfig(
+  configBuilder: ConfigOptions => HaulConfig | HaulConfig
+) {
+  return (options: ConfigOptions) => {
+    if (typeof configBuilder === 'function') {
+      return configBuilder(options);
+    }
+
+    return configBuilder;
```

Comment:
`createWebpackConfig` should return a valid webpack config with haul specific changes already done. Basically, most of the stuff from `makeReactNativeConfig` needs to be moved to this function.

Reply:
Still doesn't make any sense when compared with proposal - once it returns a function once an object. It's inconsistent, so the best fallback is a function.